File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int n) {
        vector<vector<int>> ans;
        vector<int> v;
        v.push_back(1);
        ans.push_back(v);
        for(int i=1;i<n;i++)
        {
            vector<int> u;
            u.push_back(1);
            for(int j=0;j<v.size()-1;j++) u.push_back(v[j]+v[j+1]);
            u.push_back(1);
            ans.push_back(u);
            v=u;
        }
        return ans;

        
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution(object):
    def generate(self, n):
        ans = []
        v = [1]
        ans.append(v)
        
        for i in range (1,n):
            u = []
            u.append(1)
            for j in range (len(v)-1):
                u.append(v[j]+v[j+1])
            u.append(1)
            ans.append(u)
            v=u

        return ans

        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        